c7n/resources/dynamodb.py

```diff
@@ -249,6 +249,70 @@ def process(self, resources):
                         % (f.exception()))
 
 
+@Table.action_registry.register('set-stream')
+class SetStream(BaseAction, StatusFilter):
+    """Action to enable/disable streams on table.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: stream-update
+                resource: dynamodb-table
+                filters:
+                  - TableName: 'test'
+                actions:
+                  - type: set-stream
+                    state: True
+                    stream_view_type: 'NEW_IMAGE'
+
+    """
+
+    valid_status = ('ACTIVE',)
+    schema = type_schema('set-stream',
+                         state={'type': 'boolean'},
+                         stream_view_type={'type': 'string'})
+    permissions = ("dynamodb:UpdateTable",)
+
+    def process(self, tables):
+        tables = self.filter_table_state(
+            tables, self.valid_status)
+        if not len(tables):
+            return
+
+        state = self.data.get('state')
+        type = self.data.get('stream_view_type')
+
+        stream_spec = {"StreamEnabled": state}
+
+        if self.data.get('stream_view_type') is not None:
+            stream_spec.update({"StreamViewType": type})
+
+        c = local_session(self.manager.session_factory).client('dynamodb')
+
+        futures = {}
+
+        with self.executor_factory(max_workers=2) as w:
+            for t in tables:
```

Comment:
here you can use list comprehension to eliminate the need to initialize futures above.  `futures = {w.submit(...}`